Deeploy/CommonExtensions/CodeTransformationPasses/MemoryAllocation.py

```
@@ -77,112 +77,84 @@ def apply(self,
 
-        filteredKeys = [key for key in localKeys if self._matchesRegex(ctxt, key)]
+                sortedBuffers.append(buffer)
+                unsortedBufferNames.remove(buffer.name)
 
-        for key in filteredKeys:
-            _buffer = ctxt.lookup(key)
-            if isinstance(_buffer, (StructBuffer, TransientBuffer)) or _buffer._users == []:
-                continue
-            if name == _buffer._users[-1]:
-                inputs.append(_buffer.name)
+            assert len(unsortedBufferNames) != lastLen, f"Circular reference detected."
```

Ruff (0.12.2)

Use of `assert` detected

(S101)